File: cv-frontend/lib/api/models/education_info_model.dart
import 'package:dart_json_mapper/dart_json_mapper.dart';

import 'education_level.dart';

@jsonSerializable
class EducationInfoModel{

  @JsonProperty(name: "start")
  final String startDate;
  @JsonProperty(name: "end")
  final String endDate;
  @JsonProperty(name: "name-of-university")
  final String universityName;
  @JsonProperty(name: "level")
  final EducationLevel level;

  EducationInfoModel(
      this.startDate, this.endDate, this.universityName, this.level);

  @override
  String toString() {
    var enumName = level.name;
    return '{'
        '"start": "$startDate",'
        '"end": "$endDate",'
        '"name-of-university": "$universityName",'
        '"level": "$enumName"}';
  }
}

File: cv-backend/src/main/java/a/gleb/cv/model/request/EducationInfoModel.java
package a.gleb.cv.model.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDate;

@Getter
@Setter
@ToString
public class EducationInfoModel {

    @JsonProperty("start")
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private LocalDate startDate;
    @JsonProperty("end")
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private LocalDate endDate;
    @JsonProperty("name-of-university")
    private String universityName;
    @JsonProperty("level")
    private EducationLevel level;

    public enum EducationLevel {
        MAGISTRACY("magistracy"), BACHELOR("bachelor"), SPECIALTY("specialty");

        public final String description;

        private EducationLevel(String description) {
            this.description = description;
        }
    }
}

